packages/web-app-files/src/components/SideBar/Shares/FileLinks.vue

```diff
@@ -261,14 +345,133 @@ export default defineComponent({
     },
 
     addNewLink() {
-      this.TRIGGER_PUBLIC_LINK_CREATE({
-        name: this.capabilities.files_sharing.public.defaultPublicLinkShareName,
-        expireDate: this.$_expirationDate.days
-          ? DateTime.now().plus({ days: this.$_expirationDate.days }).endOf('day').toISO()
-          : null
+      this.currentView = VIEW_CREATE
+    },
+
+    cancelCreation() {
+      this.currentView = VIEW_SHOW
+    },
+
+    getParamsForLink(link) {
+      let expireDate = ''
+
+      if (link.expiration) {
+        expireDate = DateTime.fromJSDate(link.expiration)
+          .setLocale(this.$language.current)
+          .endOf('day')
+          .toFormat("yyyy-MM-dd'T'HH:mm:ssZZZ")
+      }
+
+      let password
+
+      switch (link.password) {
+        // no password
+        case false:
+          password = undefined
+          break
+        // delete password
+        case '':
+          password = ''
+          break
+        // existing password
+        case true:
+          password = true
+          break
+        // adding/editing password
+        default:
+          password = link.password
+          break
+      }
+
+      return {
+        expireDate,
+        password,
+        permissions: link.permissions,
+        name: link.name,
+        ...(this.currentStorageId && {
+          spaceRef: `${this.currentStorageId}${this.highlightedFile.path}`
+        })
+      }
+    },
+
+    async createLink({ link, showError }) {
+      const params = this.getParamsForLink(link)
+
+      await this.addLink({
+        path: this.highlightedFile.path,
+        client: this.$client,
+        $gettext: this.$gettext,
+        params,
+        storageId: this.currentStorageId
+      }).catch(showError)
+
+      this.currentView = VIEW_SHOW
+
+      this.showMessage({
+        title: this.$gettext('Public link was created successfully')
       })
+    },
 
-      this.currentView = VIEW_EDIT
+    async updatePublicLink({ link, onSuccess = () => {}, onError = (e) => {} }) {
+      const params = this.getParamsForLink(link)
+
+      await this.updateLink({
+        id: link.id,
+        client: this.$client,
+        params
+      })
+        .then(onSuccess)
+        .catch((e) => {
+          onError(e)
+          console.error(e)
+          this.showMessage({
+            title: this.$gettext('Error when updating public link'),
```

Comment:
I think the wording in other error messages is something like `Failed to update link`.

Reply:
And I think you can already ditch the `public` in texts about links.